src/hist.jl

```diff
@@ -132,30 +132,30 @@ function sturges(n)  # Sturges' formula
     ceil(Integer, log2(n))+1
 end
 
-@compat abstract type AbstractHistogram{T<:Real,N,E} end
+abstract type AbstractHistogram{T<:Real,N,E} end
 
 # N-dimensional histogram object
-type Histogram{T<:Real,N,E} <: AbstractHistogram{T,N,E}
+mutable struct Histogram{T<:Real,N,E} <: AbstractHistogram{T,N,E}
     edges::E
     weights::Array{T,N}
     closed::Symbol
     isdensity::Bool
-    function (::Type{Histogram{T,N,E}}){T,N,E}(edges::NTuple{N,AbstractArray},
-                                               weights::Array{T,N}, closed::Symbol, isdensity::Bool=false)
+    function Histogram{T,N,E}(edges::NTuple{N,AbstractArray}, weights::Array{T,N},
+                              closed::Symbol, isdensity::Bool=false) where {T,N,E}
         closed == :right || closed == :left || error("closed must :left or :right")
         isdensity && !(T <: AbstractFloat) && error("Density histogram must have float-type weights")
         _edges_nbins(edges) == size(weights) || error("Histogram edge vectors must be 1 longer than corresponding weight dimensions")
         new{T,N,E}(edges,weights,closed,isdensity)
     end
 end
 
-Histogram{T,N}(edges::NTuple{N,AbstractVector},weights::AbstractArray{T,N},closed::Symbol=:default_left, isdensity::Bool=false) =
+Histogram(edges::NTuple{N,AbstractVector},weights::AbstractArray{T,N},closed::Symbol=:default_left, isdensity::Bool=false) where {T,N} =
```

Comment:
Okay, pushed a commit to shorten those. We should do a separate pass over the code here for stylistic updates in another PR.